module/Photo/view/photo/album-admin/add.phtml

```diff
@@ -93,6 +93,27 @@ $this->breadcrumbs()
     document.addEventListener("DOMContentLoaded", function() {
         Dropzone.options.imageDropzone = {
             acceptedFiles: 'image/*',
+            init: function () {
+                let uploading = false;
+
+                window.addEventListener("beforeunload", function (e) {
+                    if (uploading) {
+                        // Most modern browsers no longer support setting custom dialog messages
+                        // for this dialog, but for backwards compatibility this could be nice.
+                        const warning = "You are currently uploading files, are you sure you want to leave?";
+                        e.returnValue = warning;
+                        return warning;
```

Comment:
This works great! However, I suggest adding

```js
e.preventDefault();
```

at the top inside the `if`-statement. That is the new way of actually enabling this event. The other options, setting `e.returnValue` and returning `warning`, are deprecated (imho the `return warning;` can be removed as `e.returnValue` has been supported in most major browsers since 2017 or earlier).